openmrs/omod/src/main/webapp/profiles.jsp

```diff
@@ -2,104 +2,155 @@
-          <c:forEach var="file" items="${profiles}">
-            <fmt:formatDate value="${file.modified}" pattern="MMM dd, HH:mm" var="formattedDate"/>
-            <c:set var="formattedLine" value="${formattedDate} | ${fn:escapeXml(file.formattedName)} | ${file.formattedSize}"/>
-            <option value="${fn:escapeXml(file.name)}" class="filename"
-                    ${file.name == currentProfile ? 'selected' : ''}>
-              ${fn:replace(formattedLine, ' ', '&nbsp;')}
-            </option>
-          </c:forEach>
-        </select>
-        &nbsp;&nbsp;&nbsp;
-        <input type="submit" name="op" value="Apply" ${not authorized ? "disabled" : ""}>
-        <input type="submit" name="op" value="Download" ${not authorized ? "disabled" : ""}>
-        <input type="submit" name="op" value="Delete" ${not authorized ? "disabled" : ""}
-            onclick="s = document.getElementsByTagName('select')[0]; return confirm('Delete ' + s.options[s.selectedIndex].value + '?')">
-      </form>
-    </c:otherwise>
-  </c:choose>
+    <c:choose>
+        <c:when test="${empty profiles}">
+            No profiles are available.
+        </c:when>
+        <c:otherwise>
+            <form method="post" id="profile-form">
+                Select a profile:
+                <select name="profile" id="profile-select" size="3">
+                    <c:forEach var="file" items="${profiles}">
+                        <fmt:formatDate value="${file.modified}" pattern="MMM dd, HH:mm" var="formattedDate"/>
+                        <c:set var="formattedLine" value="${formattedDate} | ${fn:escapeXml(file.formattedName)} | ${file.formattedSize}"/>
+                        <option value="${fn:escapeXml(file.name)}" class="filename" ${file.name == currentProfile ? 'selected' : ''}>
+                            ${fn:replace(formattedLine, ' ', '&nbsp;')}
+                        </option>
+                    </c:forEach>
+                </select>
+                &nbsp;&nbsp;&nbsp;
+                <input type="submit" name="op" value="Apply">
+                <input type="submit" name="op" value="Download">
+                <input type="submit" name="op" value="Delete">
+            </form>
+        </c:otherwise>
+    </c:choose>
 </div>
 
+<script type="text/javascript">
+    (function() {
+        var profileQtd = ${fn:length(profiles)};
+        if(profileQtd > 3){
+            document.forms["profile-form"]["profile"].size = profileQtd;
+        }
+
+        var authorized = ${authorized};
+        if (!authorized) {
+            for (i = 0; i < document.forms.length; i++) {
+                for (j = 0; j < document.forms[i].length; j++) {
+                    document.forms[i][j].disabled = true;
+                }
+            }
+        }
+
+        document.forms["upload"]["file"].onchange = function(event){
+            document.forms['upload'].submit();
+        };
+
+        document.forms["profile-form"].onsubmit = function(event){
```

Comment:
This pops up the confirmation prompt for all the buttons (Add, Download, Delete).  Instead of attaching the onsubmit handler in a separate script, how about putting an `onsubmit` attribute on just the `<input>` element for the Delete button?

Space between `)` and `{`